src/olympia/addons/models.py

```diff
@@ -1775,6 +1775,12 @@ def set_tag_list(self, new_tag_list):
                 tag.remove_tag(self)
         self.tag_list = new_tag_list
 
+    def update_all_due_dates(self):
+        if self.current_version:
```

Comment:
That doesn't seem sufficient - that handles removing `due_date` once the add-on has been approved, assuming this is executed after the new `current_version` has been set, but wouldn't work for any other case with listed versions, right ? In particular I'm thinking about rejections, or flags changing while the add-on is still waiting for review

Reply:
Which doesn't seem ideal - we don't want to have to go through all versions every time.
- When we are adding a due date, it only needs to happen to a single version (the latest waiting for review in the right channel(s)).
- When we are removing a due date (which I guess we need to go through when adding one too, to remove the old one from the old versions), we only need to remove it to versions that had one

Right ?

Reply:
Right.  Currently the logic to clear or set is located in `reset_due_date`, that only works on a single version, so that'd need changing.